restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -145,20 +141,8 @@ public boolean treatBooleansAsNumeric() {
     return !dataStore.supportsSecondaryIndex();
   }
 
-  public static List<String> getForbiddenCharactersMessage() {
-    return forbiddenCharacters.stream().map(ch -> "`" + ch + "`").collect(Collectors.toList());
-  }
-
-  public static boolean containsIllegalChars(String x) {
-    return forbiddenCharacters.stream().anyMatch(ch -> x.indexOf(ch) >= 0);
-  }
-
-  public static String replaceIllegalChars(String x) {
-    String newStr = x;
-    for (Character y : forbiddenCharacters) {
-      newStr = newStr.replace(y, '_');
-    }
-    return newStr;
+  public static boolean containsIllegalSequences(String x) {
+    return x.contains("[") || x.contains(".") || x.contains("'");
```

Comment:
Why are `.` and `'` illegal?

As for `[` I think we can be less stringent and disallow a smaller sub-set of fields names as in regex `^\[[0-9]+\]$`, but allow something like `abc[def`. WDYT?

Reply:
One quick note on this: as a general rule, it is much easier to relax restrictions than add new ones -- so I think it'd be best to be cautious about what to allow, maybe even reserve certain characters just in case.
And definitely needs to be well documented: I am not sure what is the actual state of Documents API wrt alpha/beta/GA, but at some point these cannot be easily changed by us.

Reply:
+1 to the point on adding restrictions.

However, I really hope that we do not have to forbid any names valid in JSON document bodies. I hope all on the escaping can be on the query parameters side, where we have more control over the data structures.

The array index case seems to be the only point of contention, as far as I can see, because that is the only case where we have to decide whether to interpret the `pNNN` column value as a "field" or as an "index".

Reply:
To answer the question about `[`, `.`, and `'`, they can't be present because internally right here (https://github.com/stargate/stargate/pull/1105/files#diff-caf69767609da0fee30d28b2b410ba5e400ed01e1f42af082f495ef82fec8416R97) JSON paths like `$.a.b.c` get converted to `$['a']['b']['c']`. The point of this conversion is to allow nearly any character to be present without need to escape (this code has existed since virtually the beginning). But that means that if you have literal `'` or `[` in your paths then the conversion creates malformed paths like `$['Eric's']`, etc. Periods need escaping over the wire for a similar reason, since if the input path is `$.a......b.c` or something, you'll end up with problems in this conversion as well

Reply:
Hmmm... it might be an overkill, but could we avoid using JSON paths so as to reduce the need for escaping and allow a larger set of chars? Could we perform the same operation "manually" using simpler APIs? I suppose we only use a small sub-set of the JSON path functionality :thinking: WDYT?